### PR TITLE
Fix jump to future dates

### PR DESCRIPTION
**Pull request by Aman-Godara**
Fixes: [20800](https://github.com/vector-im/element-web/issues/20800)

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix jump to future dates ([\#8013](https://github.com/matrix-org/matrix-react-sdk/pull/8013)). Contributed by @Aman-Godara.

### CODE REVIEW

File: src/components/views/messages/JumpToDatePicker.tsx [hunk 1]
```diff
@@ -85,6 +85,7 @@ const JumpToDatePicker: React.FC<IProps> = ({ ts, onDatePicked }: IProps) => {
                 onInput={onDateValueInput}
                 onKeyDown={onDateInputKeyDown}
                 value={dateValue}
+                max={(new Date()).toISOString().substring(0, 10)}
```
Comment (MadLittleMods) [inline src/components/views/messages/JumpToDatePicker.tsx:88]: In order to craft `dateDefaultValue` above we do something a bit different. We should align whatever we choose here with that.

 1. `new Date().toISOString().substring(0, 10)`
    - This does not work when the year is `99000`: (`new Date('99000-05-05').toISOString().substring(0, 10)` -> `'+099000-05'`)
    - This also feels ripe for the same problem that user-agent parsers had when Chrome went from `99` to `100`
 1. `new Date().toISOString().split('T')[0]`
    - This does better when the year is `99000` but it's not in the format that the date `<input>` wants: `new Date('99000-05-05').toISOString().split('T')[0]` -> `'+099000-05-05'`
 
Those options are considerably simpler than the current code but this always does the right thing. We probably just need to package this logic up into a function at the top called `formatDateForInput` and re-use it in both places.

```js
    const date = new Date(ts);
    const year = date.getFullYear();
    const month = `${date.getMonth() + 1}`.padStart(2, "0");
    const day = `${date.getDate()}`.padStart(2, "0");
    const dateDefaultValue = `${year}-${month}-${day}`;
```

File: src/components/views/messages/JumpToDatePicker.tsx [hunk 2]
```diff
@@ -85,6 +85,7 @@ const JumpToDatePicker: React.FC<IProps> = ({ ts, onDatePicked }: IProps) => {
                 onInput={onDateValueInput}
```
Comment (MadLittleMods) [inline src/components/views/messages/JumpToDatePicker.tsx:85]: @Aman-Godara Is this PR meant to be in draft? Draft means you're still working on it and it's not ready for review yet.

Comment (Aman-Godara) [reply src/components/views/messages/JumpToDatePicker.tsx:85]: I was trying to test this on my local machine. The coding part is done unless testing fails.

**Comment by Aman-Godara** [comment]
Closing PR

**Pull request closed without merging**